tests/suites/test_suite_ssl.function

```diff
@@ -448,6 +453,7 @@ void ssl_message_queue_sanity()
 {
     mbedtls_test_ssl_message_queue queue;
 
+    USE_PSA_INIT();
```

Comment:
This might end up calling `mbedtls_test_ssl_message_queue_free()` on an uninitialized structure. We don't have an `_init()` function for that structure. Perhaps we should just use `mbedtls_test_ssl_message_queue queue = {0};`. Can you ask for opinions on the #ext channel?

Reply:
Fixed also in https://github.com/Mbed-TLS/mbedtls/pull/7461

Reply:
Ah, even though `{0}` should zero-initialize everything, some compilers throw a warning about it (as the CI points out), because they think perhaps we meant to provide an initializer for each field and forgot the others. I should have known, because just the other day I was reading about C23 and it's introducing `{}` as a new way to zero-initialize everything [in part to avoid such issues](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2900.htm).

Obviously we can't just start requiring C23 right now, so we'll have to find something else. I suggest defining a macro `MBEDTLS_TEST_SSL_MESSAGE_QUEUE_INIT` to `{NULL, 0, 0, 0}` near the definition of the `mbedtls_test_ssl_message_queue` type and using it here.

Reply:
Indeed the compiler I have on my workstation didn't complain about `{0}`, but then I saw the CI did, so I run the test again this morning with Docker and I updated the last commit with your proposal  ;)

Reply:
`gcc -Wall -Wextra` is happy with the `{0}` idiom to initialize all fields to logical zero, but `clang -Wall -Wextra` isn't. That's why we have macro initializers for everything in PSA that are guaranteed to be equivalent to `{0}` (or `{}` where supported), but won't trigger compiler warnings (in a good-quality implementation).